Add 2FA using allauth

davidfischer opened the pull request:
Most of the work here is templating and ensuring the settings are correct. There's no 2FA mandate and this doesn't enable webauthn/passkeys for now, strictly TOTP and recovery codes.

davidfischer commented on config/settings/base.py:
```diff
@@ -70,7 +70,7 @@
     "allauth",
     "allauth.account",
-    "allauth.socialaccount",
+    "allauth.mfa",
```
We don't plan to allow signin with github or any other social provider currently. The main users of the ad server are marketers who may not have github accounts.

ericholscher commented on adserver/templates/adserver/base.html:
```diff
@@ -42,6 +42,10 @@
             </a>
+            <a class="dropdown-item" href="{% url 'mfa_index' %}">
+              <span class="fa fa-unlock-alt fa-fw mr-2 text-muted" aria-hidden="true"></span>
+              <span>{% trans 'MFA' %}</span>
```
This should probably be spelled out? 

```suggestion
              <span>{% trans 'Multi-factor authentication (MFA)' %}</span>
```

davidfischer replied:
This doesn't fit here but I spelled out "Two-factor Auth" and then defined the acronym on the 2FA detail screen.

ericholscher commented on templates/mfa/index.html:
```diff
@@ -0,0 +1,69 @@
+    {% if authenticators.totp %}
+      <p>
+        {% blocktrans trimmed %}
+          Two-factor authentication using an authenticator app is enabled.
```
Are we calling it 2FA or MFA? We should try to use the same language.

davidfischer replied:
Sounds good. I'll make sure this fits. If it doesn't, I'll just be consistent in what we're calling it.

ericholscher commented on templates/mfa/index.html:
```diff
@@ -0,0 +1,69 @@
+          Two-factor authentication using an authenticator app is enabled.
+        {% endblocktrans %}
+      </p>
+      <p class="mb-0"><a href="{{ deactivate_url }}" class="btn btn-sm btn-danger">{% trans 'Deactivate MFA' %}</a></p>
```
There's a bunch of places where we're using this acronym, but seems ambiguous for users, and never defined in the UI?

davidfischer commented:
Updated for consistent verbiage and added a section in the account form with a link to 2FA

![image](https://github.com/user-attachments/assets/cc9c6d66-2ed8-423d-8f97-02632207adf4)
